src/v0/analytics.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ *  Properties that should be found in every analytics event
+ */
+interface SharedProperties {
+  description?: string;
+}
+
+/**
+ *  Based on `name`, what data should be sent?
+ */
+export type EventTypes =
+  | {
+      name: 'load';
+      properties: {
+        duration: number;
+      };
+    }
+  | {
+      name: 'first_render';
+      properties: {
+        duration: number;
+      };
+    }
+  | {
+      name: 'rtt_graphql';
+      properties: {
+        duration: number;
+      };
+    }
+  | {
+      name: 'token_received';
+      properties: {
+        duration: number;
+      };
+    }
+  | {
+      name: 'first_render_with_data';
+      properties: {
+        duration: number;
+        rttGraphql: number;
+        load: number;
+      };
+    }
+  | {
+      name: 'click';
+      properties: {
+        planId: string;
+      };
+    };
+
+export type EventEvent = {
+  type: 'metric' | 'component-analytics';
+} & SharedProperties &
+  EventTypes;
+
+/**
+ *  Error analytics event
+ */
+export interface ErrorEvent extends SharedProperties {
+  type: 'error';
+  name: string;
+  properties: {
+    code: string;
+    message: string;
+  };
+}
+
+export interface ErrorDetail {
+  name: string;
+  code?: string;
+  message?: string;
+}
+
+export type AnalyticsEvent = ErrorEvent | EventEvent;
+
+export const endpoint = {
+  local: 'http://analytics.arigato.tools/v1/events',
+  stage: 'https://analytics.stage.manifold.co/v1/events',
+  prod: 'https://analytics.manifold.co/v1/events',
+};
+
+export interface CreateAnalytics {
+  env: 'prod' | 'stage' | 'local';
+  element: HTMLElement;
+  componentVersion: string;
+  clientId?: string;
+}
+
+export default function createAnalytics(args: CreateAnalytics) {
+  function stringifyProperties(evt: AnalyticsEvent) {
+    return {
+      ...evt,
+      source: args.element.tagName,
+      properties: Object.entries(evt.properties).reduce(
+        (properties, [key, value]) => ({ ...properties, [key]: `${value}` }),
+        { componentName: args.element.tagName }
+      ),
+    };
+  }
+
+  /**
+   * Report an error or analytics event to Manifold
+   * @param {Object} eventData Event data to send to Manifold
+   * @param {string} eventData.type 'event' or 'error'
+   * @param {string} eventData.name name_of_event (lowercase with underscores)
+   * @param {string} [eventData.description] User-readable description of this event
+   * @param {Object} eventData.properties Free-form object of event properties (different names will require different properties)
+   * @param {Object} [options] Analytics options
+   * @param {string} [options.env] 'prod' (default) or 'stage'
+   */
+  function track(evt: AnalyticsEvent) {
```

Comment:
Again, copied over from plan-table and modified slightly so that we can close over things like `env` and `componentVersion` and `clientId`.

I chose to leave `mark` and `measure` in the plan table repo for now because it's not obvious to me that they belong here. We also probably can't access them when we actually need them if they live here. So we might look into a separate package of analytics utilities for that. cc @TerrySmithDC @domenicrosati @markjamieson